code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MultiIconPushTemplate.kt

```diff
@@ -0,0 +1,112 @@
+        if (cancelIcon.isNullOrEmpty()) {
+            cancelIcon = DEFAULT_DELETE_ICON_NAME
+        }
+    }
+
+    private fun getTemplateItemList(templateIconListJsonString: String?): MutableList<MultiIconTemplateItem>? {
+        if (templateIconListJsonString.isNullOrEmpty()) {
+            Log.warning(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Exception in converting rating uri json string to json array, Error :" +
+                    " templateIconList Json String is null or empty"
+            )
+            return null
+        }
+        val iconItemsList = mutableListOf<MultiIconTemplateItem>()
+        try {
+            val jsonArray = JSONArray(templateIconListJsonString)
+            for (i in 0 until jsonArray.length()) {
+                val jsonObject = jsonArray.getJSONObject(i)
+                val multiIconAction = getIconItemFromJsonObject(jsonObject)
+                multiIconAction?.let { iconItemsList.add(it) }
+            }
+        } catch (e: Exception) {
+            Log.debug(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Exception in converting template action json string to json array, Error : ${e.localizedMessage}"
+            )
+            return null
+        }
+        return iconItemsList
+    }
+
+    private fun getIconItemFromJsonObject(jsonObject: JSONObject): MultiIconTemplateItem? {
+        return try {
+            val imageUri = jsonObject.getString(PushTemplateConstants.MultiIconTemplateKeys.IMG)
+            // In case of invalid image URI, return null as icon is mandatory
+            if (imageUri.isNullOrEmpty()) {
+                Log.debug(
+                    PushTemplateConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Image uri is empty, cannot create icon item."
+                )
+                return null
+            }
+            var uri: String? = null
+            val actionTypeString = jsonObject.getString(PushTemplateConstants.MultiIconTemplateKeys.TYPE)
+            var actionType = if (actionTypeString.isNullOrEmpty()) {
+                PushTemplateConstants.ActionType.NONE
+            } else {
+                PushTemplateConstants.ActionType.valueOf(actionTypeString)
+            }
+            if (actionType == PushTemplateConstants.ActionType.WEBURL || actionType == PushTemplateConstants.ActionType.DEEPLINK) {
+                uri = jsonObject.getString(PushTemplateConstants.MultiIconTemplateKeys.URI)
+                if (uri.isNullOrEmpty()) {
+                    Log.debug(
+                        PushTemplateConstants.LOG_TAG,
+                        SELF_TAG,
+                        "Uri is empty for action type $actionType, cannot create icon item."
+                    )
+                    return null
+                }
+            }
+
+            MultiIconTemplateItem(imageUri, actionType, uri)
+        } catch (e: Exception) {
```

Comment:
is this catching a JSON exception? let's log the exception with a trace log to better investigate any possible issues in the future.

Reply:
Done